cli/src/main/java/org/jboss/as/cli/operation/OperationRequestCompleter.java

```diff
@@ -592,6 +601,9 @@ private int completeArgumentValueAndPropertyNames(CommandContext ctx, ParsedComm
                 final CommandLineFormat format = parsedCmd.getFormat();
                 if (format != null && format.getPropertyListEnd() != null && format.getPropertyListEnd().length() > 0) {
                     candidates.add(format.getPropertyListEnd());
+                    if (candidates.size() == 1) {
+                        result = parsedCmd.getLastChunkIndex() + chunk.length();
```

Comment:
It should be ensured by line https://github.com/wildfly/wildfly-core/commit/11911af7eea6dfd89c8cb2f843524a8c75230282#diff-f91220412cef158ff6623a2dccc627a4R462 
where the lastArg is only populated in case the FullName of the arg is not null (.equals would fail with NPE if it was called on null) and is equal to the chunk and the line
https://github.com/wildfly/wildfly-core/commit/11911af7eea6dfd89c8cb2f843524a8c75230282#diff-f91220412cef158ff6623a2dccc627a4R554
where it is checked that lastArg is not null.

However, the lastArg and chunk are mutable, therefore in case there would be committed a code which assigns a null to either of them, this indirect "check" would fail.
Do You think it would be better to add a null check of the chunk in the new method recalculateResult() commited in 11911af ?

Reply:
It would be nice to add a check and a comment.

Reply:
Thank you very much, I have added them.